### ISSUE TITLE
Questions about ChatGLM3-6b-128k

### ISSUE THREAD

**Issue by KaiLv69** [THUDM/LongAlign#2]
Hello, I am very interested in your work, but I have a few questions below.

1. How to further train a chat model like ChatGLM3-6b? As we know that they have templates, will the further training affect the instruction-following ability or any unstable issues?

2. How to get ChatGLM3-6b-128k? From the paper I find that ChatGLM3-6b-64k is trained from ChatGLM3-32k with 10b further training data and SFT with longalign. Maybe I missed how to get ChatGLM3-6b-128k.

**Comment by bys0318** [comment]
Hi! As we stated in sec 4.4 of our paper, "We also construct SFT data up to 128k in length with human annotation and successfully align ChatGLM3-6B under 128k context window using packing training with loss weighting, resulting in ChatGLM3-6B-128k". The 128k-length training data will be made public in the future. Currently, we release the 64k-length LongAlign-10k dataset at https://huggingface.co/datasets/THUDM/LongAlign-10k.
Regarding Q1, the long instruction training data also follows the original ChatGLM3-6b chat prompt, and we have not yet spotted a degradation on its instruction-following ability after LongAlign.

**Comment by KaiLv69** [comment]
In my understanding, did you use 10 billion tokens of long instruction training data to train ChatGLM3-6B?

**Comment by bys0318** [comment]
No. 10 billion tokens is the amount of long pretrained data for context scaling (continual training). The long instruction training data (SFT) for long context alignment is much less.

**Comment by bys0318** [comment]
Refer to sec 4.1 in our paper ("model" and "training") for the full process.

**Comment by KaiLv69** [comment]
Let's clarify this: 
To get Chatglm3-6b-128k, you first use 10b pre-training data (**q1**: this step probably didn't have a template?) to expand to 128k (**q2**: 128k or 64k?).
Then, you use 128k-length data for SFT (with templates).

**Comment by bys0318** [comment]
Exactly. q1: no this step does not involve any template; q2: 128k

**Comment by KaiLv69** [comment]
Thx. Happy new year! :-)
